magefile.go

```diff
@@ -542,17 +540,81 @@ func (Test) Cluster() error {
 
 	// Build test binary with ginkgo for better output
 	fmt.Println("🔨 Building test binary with ginkgo...")
-	err = sh.RunWith(map[string]string{
+
+	// Prepare environment variables for test build
+	buildEnv := map[string]string{
 		"CGO_ENABLED": "1",
-	}, "ginkgo", "build", "./tests/e2e/")
+	}
+
+	// Add SQUID_REPLICA_COUNT if specified, so it's available during BeforeSuite
```

Comment:
Do we also need to account for assigning the env vars when the value is taken from the values file?